### PR TITLE
GEODE_4132: Sending handshake response from locator for new protocol

### PR DESCRIPTION
**Pull request by upthewaterspout**
Changing the locator to send a response to the NewConnectionVersion
message to be consistent with the server. This change also allows the
locator to handle an authentication message, which means that
GetAvailableServers will work if a security manager is enabled and the
client sends a validate authentication message.

Signed-off-by: Sarge <mdodge@pivotal.io>

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

@PivotalSarge @WireBaron

### CODE REVIEW

File: geode-protobuf/src/test/java/org/apache/geode/internal/protocol/protobuf/v1/acceptance/LocatorConnectionAuthenticationDUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    Host host = Host.getHost(0);
+    Socket socket = new Socket(host.getHostName(), locatorPort);
+    MessageUtil.sendHandshake(socket);
+    MessageUtil.verifyHandshakeSuccess(socket);
+    return socket;
+  }
+
+  /**
+   * Test that if the locator has a security manager, an unauthorized client is not allowed to do
+   * anything.
```
Comment (WireBaron) [inline geode-protobuf/src/test/java/org/apache/geode/internal/protocol/protobuf/v1/acceptance/LocatorConnectionAuthenticationDUnitTest.java]: This comment seems wrong

File: geode-protobuf/src/test/java/org/apache/geode/internal/protocol/protobuf/v1/acceptance/LocatorConnectionAuthenticationDUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+   */
+  @Test
+  public void authorizedClientCanGetServersIfSecurityIsEnabled() throws Throwable {
+    ClientProtocol.Request.Builder protobufRequestBuilder =
+        ProtobufUtilities.createProtobufRequestBuilder();
```
Comment (WireBaron) [inline geode-protobuf/src/test/java/org/apache/geode/internal/protocol/protobuf/v1/acceptance/LocatorConnectionAuthenticationDUnitTest.java]: Doesn't look like this is used anywhere

File: geode-protobuf/src/test/java/org/apache/geode/internal/protocol/protobuf/v1/acceptance/LocatorConnectionAuthenticationDUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+          protobufProtocolSerializer.deserialize(socket.getInputStream());
+      assertEquals(true,
+          authorizationResponse.getResponse().getAuthenticationResponse().getAuthenticated());
+      protobufProtocolSerializer.serialize(getAvailableServersRequestMessage,
+          socket.getOutputStream());
+
+      ClientProtocol.Message getAvailableServersResponseMessage =
+          protobufProtocolSerializer.deserialize(socket.getInputStream());
+      assertNotNull("Got response: " + getAvailableServersResponseMessage,
+          getAvailableServersRequestMessage.getResponse().getErrorResponse());
```
Comment (WireBaron) [inline geode-protobuf/src/test/java/org/apache/geode/internal/protocol/protobuf/v1/acceptance/LocatorConnectionAuthenticationDUnitTest.java]: Do we expect an error response here?  I'd have thought the test should be sending a getAvailableServersResponse after we authenticated.

Comment (upthewaterspout) [reply geode-protobuf/src/test/java/org/apache/geode/internal/protocol/protobuf/v1/acceptance/LocatorConnectionAuthenticationDUnitTest.java]: You are right. Looks like we are still getting an error response!

**Pull request merged**